pandas/tests/indexes/datetimes/test_date_range.py

```diff
@@ -179,6 +179,18 @@ def test_date_range_convenience_periods(self):
                              '2018-04-01 03:00:00+10:00',
                              '2018-04-01 04:00:00+10:00'], freq=None)
 
+    @pytest.mark.parametrize('start,end', [
+        ['20180101', '20180103'],
+        [datetime(2018, 1, 1), datetime(2018, 1, 3)],
+        [Timestamp('20180101'), Timestamp('20180103')],
+        [Timestamp('20180101', tz='US/Eastern'),
+         Timestamp('20180103', tz='US/Eastern')]])
+    def test_date_range_linspacing_tz(self, start, end):
+        # GH 20983
+        result = date_range(start, end, periods=3, tz='US/Eastern')
```

Comment:
Might be beneficial to test when `start`/`end` are tz aware but not specifying the `tz` param for `date_range`, to ensure that `tz` inference from `start`/`end` is okay.